wmidump.c

```diff
@@ -174,34 +174,37 @@ static void *parse_ascii_wdg(const char *wdg, size_t *bytes)
 	return data;
 }
 
-static void *read_wdg(int fd, size_t *len)
+static char *read_wdg(int fd, size_t *len)
 {
-	void *data = NULL;
-	char buf[1024];
+	char buf[BUFSIZ];
+	char *data = NULL;
 	ssize_t bytes;
 
 	*len = 0;
 
 	while ((bytes = read(fd, buf, sizeof(buf))) > 0) {
-		size_t offset = *len;
+		data = realloc(data, *len + bytes);
+		if (data == NULL)
+			err(1, NULL);
+		memcpy(data + *len, buf, bytes);
 		*len += bytes;
-		data = realloc(data, *len);
-		memcpy((uint8_t *)data + offset, buf, bytes);
 	}
 	if (bytes < 0) {
-		perror("read()");
+		warn("read");
```

Comment:
This really isn't part of a refactor. Changing the logging functions should be a separate change - and it should come with a justification. Why do it?